pilot/pkg/model/authentication.go

```diff
@@ -312,7 +313,24 @@ func SdsCertificateConfigFromResourceName(resource string) (SdsCertificateConfig
 	} else if strings.HasPrefix(resource, "file-root:") {
 		filesString := strings.TrimPrefix(resource, "file-root:")
 		split := strings.Split(filesString, ResourceSeparator)
-		if len(split) != 1 {
+		if filesString == "system" && len(split) != 1 {
+			var certFiles = []string{
```

Comment:
I feel like this needs to be under some agent/ folder. I was pretty surprised to find SdsCertificateConfigFromResourceName was only ever called from the agent currently, which makes me think its fairly fragile and may lead to an import from Istiod, which would be incorrect.

Reply:
This is an exported method that doesn't have any dependencies within the package. I'd consider moving it to a file within a folder used by the agent. If someone else needs this functionality at a later time, it could be moved to a shared folder.